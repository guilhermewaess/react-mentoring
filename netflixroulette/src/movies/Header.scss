// .movies-header{
//   margin-bottom: 0;
// }