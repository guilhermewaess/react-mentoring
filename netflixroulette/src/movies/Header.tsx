import * as React from 'react';
import { Jumbotron } from 'reactstrap';
import './Header.scss';

export class Header extends React.Component {
  public render() {
    return (
      <Jumbotron fluid={true} className="movies-header">
          <h1>TESTE</h1>
      </Jumbotron>
    );
  }
}
