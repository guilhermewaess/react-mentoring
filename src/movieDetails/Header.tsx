import * as React from 'react';
import { Link } from 'react-router-dom';
import { Badge, Button, Card, CardText, CardTitle, Col, Container, Row } from 'reactstrap';
import './Header.scss';

export default (props: { movie: any }) => {
  return (
    <Container fluid={true} className="movie-details-header">
      <Row className="h-100">
        <Col xs={3}>
          <img
            src={props.movie.poster_path}
            className="rounded mx-auto d-block p-10"
            alt="poster"
          />
        </Col>
        <Col xs={8} className="align-self-center">
          <Card body={true} className="movie-detail-data text-white border-0">
            <CardTitle className="mb-0">
              {props.movie.title}{' '}
              <Badge color="secondary">Score: {props.movie.vote_count}</Badge>
            </CardTitle>
            <h6 className="movie-release-date">{props.movie.release_date}</h6>
            <CardText>{props.movie.overview}</CardText>
          </Card>
        </Col>
        <Col xs={1}>
          <Link to="/">
            <Button
              outline={true}
              className="search-button 
                         mt-1"
            >
              Search
            </Button>
          </Link>
        </Col>
      </Row>
    </Container>
  );
};
