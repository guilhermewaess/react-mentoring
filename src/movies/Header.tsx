import * as React from 'react';
import { Button, Col, Container, Row } from 'reactstrap';
import './Header.scss';
import SearchBy from './SearchBy';

const SearchInput = (props: any) => (
  <div>
    <label className="mb-1 font-weight-bold">Find your movie</label>
    <div className="shadow-lg search-input">
      <input
        data-filter-field="search"
        onChange={props.handleFilterChange}
        className="w-100
                border-0
                rounded"
        type="text"
        placeholder="Search your movie"
      />
    </div>
  </div>
);

export default (props: any) => {
  return (
    <Container fluid={true} className="movies-header">
      <div className="text-white-50">Netflixroulette</div>
      <Row className="h-100">
        <Col
          xs={{ offset: 2, size: 8 }}
          className="align-self-center text-white"
        >
          <form onSubmit={props.onSearch}>
            <SearchInput handleFilterChange={props.handleFilterChange} />
            <SearchBy
              filter={props.filter}
              handleFilterChange={props.handleFilterChange}
            />
            <Button
              outline={true}
              className="search-button 
                         mt-1
                         float-right"
            >
              Search
            </Button>
          </form>
        </Col>
      </Row>
    </Container>
  );
};
