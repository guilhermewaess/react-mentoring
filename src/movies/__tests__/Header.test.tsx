import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import Header from './../Header';

describe('Movies Header', () => {
  let component: ShallowWrapper;

  beforeEach(() => {
    component = shallow(<Header />)
  });

  describe('when construct', () => {
    it('should render component', () => {
      expect(shallowToJson(component)).toMatchSnapshot();
    });
    it('should render SearchInput properly', () => {
      const searchInput = component.find('SearchInput').shallow();
      expect(shallowToJson(searchInput)).toMatchSnapshot();
    });
  });
});