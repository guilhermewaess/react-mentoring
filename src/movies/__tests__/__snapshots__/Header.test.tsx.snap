// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies Header when construct should render SearchInput properly 1`] = `
<div>
  <label
    className="mb-1 font-weight-bold"
  >
    Find your movie
  </label>
  <div
    className="shadow-lg search-input"
  >
    <input
      className="w-100
                border-0
                rounded"
      placeholder="Search your movie"
      type="text"
    />
  </div>
</div>
`;

exports[`Movies Header when construct should render component 1`] = `
<Container
  className="movies-header"
  fluid={true}
  tag="div"
>
  <div
    className="text-white-50"
  >
    Netflixroulette
  </div>
  <Row
    className="h-100"
    tag="div"
  >
    <Col
      className="align-self-center text-white"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs={
        Object {
          "offset": 2,
          "size": 8,
        }
      }
    >
      <SearchInput />
      <Button
        className="search-button 
                         mt-1
                         w-20
                         float-right"
        color="secondary"
        outline={true}
        tag="button"
      >
        Search
      </Button>
    </Col>
  </Row>
</Container>
`;
